src/guide/advanced/authentication.md

```diff
@@ -200,16 +208,186 @@ To request a protected route, you will need to add the following header:
 Authorization: Bearer <YOUR_TOKEN>
 ```
 
+##### Optional authentication
+
 This provider supports the `Optional` flag, which defines if the authenticator allows requests that don't provide credentials. Handlers should therefore check if `request.User` is not `nil` before accessing it.
 
 ``` go
 authenticator := auth.Middleware(&model.User{}, &auth.JWTAuthenticator{Optional: true})
 router.Middleware(authenticator)
 ```
 
+##### Custom ID claim name
+
+By default, `auth.JWTAuthenticator` looks for the claim named `userid` in the given token. You can customize the name of the token using the `ClaimName` field:
+
+``` go
+authenticator := auth.Middleware(&model.User{}, &auth.JWTAuthenticator{Optional: true})
```

Comment:
It seems this example is a copy/past from the previous category.
I think it missing the struct example which show how to use ClaimName, am i right ?

Reply:
Good catch, thank you!